packages/core-utils/src/optimism/batch-encoding.ts

```diff
@@ -38,12 +81,39 @@ export const encodeAppendSequencerBatch = (
     ).padStart(6, '0')
     return acc + encodedTxDataHeader + remove0x(cur)
   }, '')
-  return (
+
+  if (b.type === BatchType.ZLIB) {
+    const compressed = zlib
+      .deflateSync(Buffer.from(encodedTransactionData, 'hex'))
+      .toString('hex')
+
+    encodedTransactionData = compressed
+    contexts = [
+      {
+        numSequencedTransactions: 0,
+        numSubsequentQueueTransactions: 0,
+        timestamp: 0,
+        blockNumber: 0,
+      },
+      ...contexts,
+    ]
+  }
+
+  const encodedContextsHeader = encodeHex(contexts.length, 6)
+  const encodedContexts =
+    encodedContextsHeader +
+    contexts.reduce((acc, cur) => acc + encodeBatchContext(cur), '')
+
+  const ret =
     encodeShouldStartAtElement +
     encodedTotalElementsToAppend +
     encodedContexts +
     encodedTransactionData
-  )
+
+  if (opts?.buffer) {
+    return Buffer.from(remove0x(ret), 'hex')
```

Comment:
JS ethereum tooling likes strings, ideally all strings are a `Buffer` or `uint8array` in memory and then there are `toJSON` and `fromJSON` methods on the types that convert buffers to strings. Dealing with strings is annoying because there has to be `remove0x` or `add0x` everywhere when manipulating them as well as the overhead of needing to copy the whole thing each time its sliced. Shoutout to [bcoin](https://github.com/bcoin-org/bcoin/blob/ff7e43bfd935b778e8d01017470007b9d609a6af/lib/primitives/tx.js#L2155)/[hsd](https://github.com/handshake-org/hsd/blob/df997a4c305eb8e730fefbaf12bf5bc041293aa3/lib/primitives/block.js#L359).

The batch encoding function should accept a string or buffer and be optimized to operate solely on buffers with the option to return it as a buffer or string. I wanted to get this PR in sooner and optimize later